x-pack/plugins/fleet/server/services/epm/packages/install.ts

```diff
@@ -97,22 +97,50 @@ export async function ensureInstalledDefaultPackages(
   });
 }
 
+export async function isPackageInstalled(options: {
+  savedObjectsClient: SavedObjectsClientContract;
+  pkgName: string;
+  version?: string;
+}): Promise<Installation | false> {
+  const { savedObjectsClient, pkgName, version } = options;
+  const installedPackage = await getInstallation({ savedObjectsClient, pkgName });
+  if (installedPackage && (!version || installedPackage.version === version)) {
+    return installedPackage;
+  }
+  return false;
+}
+
 export async function ensureInstalledPackage(options: {
   savedObjectsClient: SavedObjectsClientContract;
   pkgName: string;
   esClient: ElasticsearchClient;
+  version?: string;
 }): Promise<Installation> {
-  const { savedObjectsClient, pkgName, esClient } = options;
-  const installedPackage = await getInstallation({ savedObjectsClient, pkgName });
+  const { savedObjectsClient, pkgName, esClient, version } = options;
+  const installedPackage = await isPackageInstalled({ savedObjectsClient, pkgName, version });
   if (installedPackage) {
     return installedPackage;
   }
   // if the requested packaged was not found to be installed, install
-  await installLatestPackage({
-    savedObjectsClient,
-    pkgName,
-    esClient,
-  });
+  if (version) {
+    const pkgkey = Registry.pkgToPkgKey({
+      name: pkgName,
+      version,
+    });
+    await installPackage({
+      installSource: 'registry',
+      savedObjectsClient,
+      pkgkey,
+      esClient,
+      force: true,
```

Comment:
hmm, passing this force flag worries me a little. a user could set an older version of a package (than what they currently have installed) in their preconfiguration and this would allow them to install that, I'll think about this some more but am interested in hearing your thoughts too

Reply:
Should we change `isPackageVersionInstalled` so that it returns true if `pkgVersion` or later is installed?

Reply:
let's leave the force flag here for now, otherwise there would be no point in accepting a `version` param

@ruflin can you share some thoughts on why we want the user to define an explicit version in their preconfigured packages/policies? is there a reason that we allow them to downgrade from what they already have installed, or install a new package that is *not* the latest available on EPR?

Reply:
With this API the user should be able to setup the same environment multiple times. The API today on a fresh installation of Kibana 7.11.2 should lead to the same result in two weeks for 7.11.2.

The question is what we should do, if a newer version is already installed. Few ideas:

* Return an error as it is not compatible
* Introduce a force flag on the new API to also allow this
* Downgrade (this might have side effects)

We should work out and document the detailed behaviour.